sdk/nodejs/runtime/resource.ts

```diff
@@ -510,7 +510,11 @@ async function prepareResource(label: string, res: Resource, custom: boolean, re
 
         // Serialize out all our props to their final values.  In doing so, we'll also collect all
         // the Resources pointed to by any Dependency objects we encounter, adding them to 'propertyDependencies'.
-        const [serializedProps, propertyToDirectDependencies] = await serializeResourceProperties(label, props);
+        const [serializedProps, propertyToDirectDependencies] = await serializeResourceProperties(label, props, {
+            // To initially scope the use of this new feature, we only keep output values when
+            // remote is true (for multi-lang components).
+            keepOutputValues: remote,
```

Comment:
I've gone back-and-forth on whether the SDK should always keep output values (if the monitor supports it), leaving it to the engine to handle whether output values should be kept when unmarshaling.

However, to really scope and de-risk the change, for now, in the SDKs, if the monitor supports it, we'll only keep output values when marshaling values for `RegisterResource` when `remote` is `true` and for `Call`. (Additionally, initially, the engine will only keep output values during unmarshaling for `RegisterResource` when `remote` is true, and for `Call`). In the future, if/when we adopt output values more broadly we can update the SDKs to keep output values in more cases, and similarly have the engine do the same.

Reply:
The concern is roll-out order? So basically we need all our clients to update to new monitor that understands the extended protocol before SDKs can assume it and use the new protocol?

Reply:
> The concern is roll-out order?

No. The feature support checking will take care of this. SDKs will only send output values if the monitor says it supports them, and then engine will only pass output values to a provider if it says it accepts them. And even when accepting output values, if the code that's accepting the output values doesn't have `KeepOutputValues` specified when unmarshaling, it will get back the same values as previously.

I'm mainly concerned that output values will somehow show up in places that can't fully accommodate them (yet), both (1) within an SDK and (2) within the engine. Perhaps I'm being overly paranoid, but given the short timeframe for rolling this out, and being involved with a similar feature (resource references) that was enabled more broadly and had a bit of a bug tail, I'm inclined to really scope down where the feature is enabled as much as possible initially.

I'm less concerned about (1), because I've done a scrub of the places where we marshal and keep output values in the SDKs. For example, there are places in the SDK where we roundtrip inputs (i.e. marshal and then unmarshal), e.g. for mocks-based testing. In that case, since the mock monitor doesn't support the feature, it should all just work as expected.

Slightly more concerned about (2). When unmarshaling the RPC values, if `KeepOutputValues` isn't true, we should get the exact same behavior as before (any output values will be unmarshaled in the old way) so we should be covered. However, if somehow output values _did_ show up, such as enabling it for _all_ `RegisterResource` calls, I'm concerned the output values might somehow mess up areas that haven't been looked at closely, such as diffs, where previously `resource.Output` was treated as unknown.

Again, these concerns are theoretical because I don't think we actually have any problems here, I'm just aiming to reduce risk.